core/controllers/ingest.py

```diff
@@ -82,6 +83,20 @@ def clean_data(workbook: dict[str, pd.DataFrame]) -> None:
         worksheet.replace({np.nan: None}, inplace=True)  # replaces np.NAT with None
 
 
+def next_submission_id(reporting_round: int) -> str:
+    """Get the next submission ID by incrementing the last in the DB.
+
+    :return: The next submission ID.
+    """
+    submissions = Submission.query.filter_by(reporting_round=reporting_round).all()
+    if not submissions:
+        return SUBMISSION_ID_FORMAT.format(reporting_round, 1)  # the first submission
+
+    latest_submission = sorted(submissions, key=lambda x: x.submission_number)[0]
```

Comment:
I think you should be able to order the data in the original query, with something like `Submission.query.filter_by(reporting_round=reporting_round).order_by(Submission.submission_number).all()`

Reply:
Or even better, sort and just return the top value, something like:
`Submission.query.filter_by(reporting_round=reporting_round).order_by(desc(Submission.submission_number)).first()`

Reply:
I tried this originally but it doesn't look like you can query on hybrid parameters like this. I imagine because the query is executed at SQL level but the hybrid param is at Python/ORM level? I may have been implementing it slightly wrong though so will give it another go

Reply:
yeah this errors....

Reply:
ah ok, shame

Reply:
as discussed, try ordering on Submission_id itself? should work on a string